arelle/ModelDocument.py

```diff
@@ -1164,6 +1164,8 @@ def linkbaseDiscover(self, linkbaseElement, inInstance=False):
                                             linkrole = XbrlConst.defaultLinkRole
                                         #index by both arcrole and linkrole#arcrole and dimensionsions if applicable
                                         baseSetKeys = [(arcrole, linkrole, linkQn, arcQn)]
+                                        baseSetKeys.append((arcrole, linkrole, linkQn, None))
```

Comment:
Pre-existing/unrelated I think it would make it significantly easier to reason about if we made objects for these keys instead of using tuples